### PR TITLE
JENKINS-56133:  Add build url as default annotation

### PR DESCRIPTION
**Pull request by MattLud**

### CODE REVIEW

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesTestUtil.java [hunk 1]
```diff
@@ -142,9 +142,39 @@ public static void assumeKubernetes() throws Exception {
         return l;
     }
 
+    public static Map<String, String> getAnnotations(KubernetesCloud cloud, String podName) {
```
Comment (carlossg) [inline src/test/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesTestUtil.java]: you are not using this method

Comment (MattLud) [reply src/test/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesTestUtil.java]: It's being used in KubernetesPipelineTest and KubernetsDeclarativeAgent test - 
image

Comment (carlossg) [reply src/test/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesTestUtil.java]: yes, you are using it in the tests but not in the actual plugin, there's nothing that will add the annotations to the pods when they are launched

Comment (MattLud) [reply src/test/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesTestUtil.java]: The annotation is being added automatically by the step execution - https://github.com/jenkinsci/kubernetes-plugin/pull/433/files#diff-5501a99a43da60cbad908396dfee9336R100 - the method is necessary to confirm that the the annotation was added and properly reflects the build url.

Comment (carlossg) [reply src/test/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesTestUtil.java]: my bad, missread it

**Comment by MattLud** [comment]
Is there a link to the failed tests? Worked fine locally, but I'm worried it was an edge case.

**Comment by MattLud** [comment]
Going to rewrite the tests to match runInPod approach to verifying to avoid race condition.
